starknet_devnet/blueprints/base.py

```diff
@@ -221,3 +221,37 @@ async def fork_status():
     if config.fork_network:
         return jsonify({"url": config.fork_network.url, "block": config.fork_block})
     return jsonify({})
+
+
+@base.route("/abort_blocks", methods=["POST"])
+async def abort_blocks():
+    """Abort blocks and transactions from given block hash to last block."""
+    request_json = request.json or {}
+    starting_block = await state.starknet_wrapper.blocks.get_by_hash(
+        hex(hex_converter(request_json, "startingBlockHash"))
+    )
+    last_block = await state.starknet_wrapper.blocks.get_last_block()
+    blocks_to_abort = []
+    aborted_blocks = []
+
+    # Get blocks to abort
+    for block_number in range(starting_block.block_number, last_block.block_number + 1):
```

Comment:
I don't know if this is the optimal way, but if you are going to want to change something, you might want to consider consider relying on `parent_block_hash` property of `StarknetBlock`. That way we could just get the latest block and iterate backwards until the starting block is reached.